clash-lib/src/Clash/Driver/Types.hs

```diff
@@ -433,6 +436,7 @@ instance NFData ClashOpts where
     opt_edalize o `deepseq`
     opt_renderEnums o `deepseq`
     opt_timescalePrecision o `deepseq`
+    opt_ignoreBrokenGhcs o `deepseq`
```

Comment:
Not really a comment on this PR, but this feels a little small to open an issue over.

Why aren't we just `deriving NFData ClashOpts`?
This manual instance doesn't evaluate `opt_debug` and `seq`s rather then `deepseq`s `opt_color`, is that by accident, or on purpose?
If so can we probably add a little comment documenting this?

Reply:
I think because `OverridingBool` doesn't have one, but I'd have to check to be sure

Reply:
It is indeed `OverridingBool`. I've added an issue https://github.com/clash-lang/clash-compiler/issues/2754.